### PR TITLE
feat: add number separators, units, conditional required, hex strings

### PR DESCRIPTION
**Pull request by Cafe137**
* Adds numerical units and underscore separator
* Adds hex-string type
* Adds `required.when`
* Replaces `--hash-topic` with `--topic-string`
* Adds `Usability Features` section to README

Requested changes:

- [x] Make identities use `required.when` and provide interactive picker there as well
- [x] Password input when identity requires it but not provided beforehand
- [x] Change address type to `hex-string`
- [ ] Add larger numbers to quality-of-life tests
- [x] Add mixed unit and underscore to quality-of-life tests

~Change feed reference options to arguments~

It conflicts with the `identity` option, so I would keep it as an option to retain this relationship

### CODE REVIEW

**Review by nugaon — COMMENTED** [review]
Nice upgrades,

but I would apply the same `required.when` for identities as well and where it is necessary I would print out a picker like in case of postage stamps.
it comes from the following experience:
`swarm-cli feed update` -> `Required option not provided: identity -> identity list, then pick one that I provide to this command -> `There is no password passed for V3 wallet initialization` (also force me to provide the password in the command instead of providing it interactively) -> provide password in the CLI command and only after then I have a successful run.

also I would apply this new `hex-string` type for `--address` of `swarm-cli feed print` and other similar places. 

off-topic question: why we want `reference` as an option instead as an argument at `swarm-cli feed update`?

File: test/quality-of-life/numerical.spec.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  })
+  process.env.SWARM_CLI_CONFIG_FOLDER = configFolderPath
+  process.env.SWARM_CLI_CONFIG_FILE = configFileName
+
+  beforeEach(() => {
+    consoleMessages.length = 0
+  })
+
+  it('should be possible with underscores and units', async () => {
+    await invokeTestCli(['stamp', 'buy', '--amount', '1M', '--depth', '16', '--gas-price', '10_000'])
```
[the remark this answers is not in the document]
Comment (Cafe137) [reply test/quality-of-life/numerical.spec.ts:20]: Good points @nugaon, I will create a checklist for your suggestions.

I will see if we can test large numbers without hitting ETH / BZZ limits.

File: src/command/pss/pss-command.ts [hunk 2]
```diff
@@ -1,23 +1,17 @@
...
-  @Option({ key: 'hash-topic', alias: 'H', type: 'boolean', description: 'Hash the topic to 32 bytes', default: false })
-  public hashTopic!: boolean
+  @Option(topicStringProperties)
+  public topicString!: string
 
   public init(): void {
     super.init()
 
-    this.topic = getTopic(this.bee, this.console, this.topic, this.hashTopic)
+    this.topic = this.topic || this.bee.makeFeedTopic(this.topicString)
```
Comment (nugaon) [inline src/command/pss/pss-command.ts:15]: So as it was discussed under [a bee-js PR](https://github.com/ethersphere/bee-js/pull/343#discussion_r648490565), there is no way currently to pass hashed topic from user side, so this part is wrong, but could be fixed later in an other PR with resolving [this](https://github.com/ethersphere/swarm-cli/issues/141) issue

**Review by nugaon — APPROVED** [review]
LGTM, but please remove `isHexString` and `verifyHex` from `utils/hex.ts` since it is not used anymore

**Pull request merged**